### PR TITLE
new scenario data queries tests

### PR DESCRIPTION
**Pull request by amuolo**
This PR adds a notebook to test query methods for scenario data, ticket #99

### CODE REVIEW

File: ifrs17/Test/QueriesScenarioDataTest.ipynb [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+            "cell_type": "code",
+            "source": [
+                "(await ws.QueryReportVariablesAsync((args.Year, args.Month, args.ReportingNode, argsScenarioMTUP.Scenario))).Select(x => x.Value).Sum().Should().Be(21.2);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
```
Comment (dcolleoni) [inline ifrs17/Test/QueriesScenarioDataTest.ipynb:375]: Remove empty cells

File: ifrs17/Test/QueriesTest.ipynb [hunk 2]
```diff
@@ -809,7 +825,70 @@
...
+                "\nerrorMessage.Contains(expectedErrorMessage).Should().Be(true);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var args = new Args(\"CH\",2020,9,Periodicity.Monthly,\"MTUP\");",
```
Comment (dcolleoni) [inline ifrs17/Test/QueriesTest.ipynb:837]: the appropriate scenario for YC would be YCUP, no?

File: ifrs17/Test/TestData.ipynb [hunk 3]
```diff
@@ -15,6 +15,15 @@
     "nbformat": 4,
     "nbformat_minor": 5,
     "cells": [
+        {
+            "cell_type": "code",
+            "source": [
+                "#!import \"../CalculationEngine\""
```
Comment (dcolleoni) [inline ifrs17/Test/TestData.ipynb:21]: I would import only what is necessary, in this way we can keep compilation time down.

**Comment by dcolleoni** [comment]
@amuolo I don't see the tests on IfrsVariable queries in the ImportStorage as described in the issue:
"IfrsVariable queries in ImportStorage : Test IfrsVariables Query => IFRS 17 Template > Test > ImportStorageTest"

File: ifrs17/Test/TestData.ipynb (hunk 3 above)
Comment (amuolo) [reply ifrs17/Test/TestData.ipynb:21]: ok agree

**Comment by amuolo** [comment]
> @amuolo I don't see the tests on IfrsVariable queries in the ImportStorage as described in the issue: "IfrsVariable queries in ImportStorage : Test IfrsVariables Query => IFRS 17 Template > Test > ImportStorageTest"

@dcolleoni added

File: ifrs17/Test/QueriesScenarioDataTest.ipynb [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+                "\nws.Initialize(x => x.FromSource(DataSource).DisableInitialization<RawVariable>().DisableInitialization<IfrsVariable>());"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var queriedRawVars = await ws.QueryPartitionedDataAsync<RawVariable,PartitionByReportingNodeAndPeriod>(DataSource, partitionScenarioMTUP.Id, partition.Id, ImportFormats.Cashflow);"
```
Comment (dcolleoni) [inline ifrs17/Test/QueriesScenarioDataTest.ipynb:243]: QueryPartitionedDataAsync behaves differently if the type is RawVariable or IfrsVariable. 
I would expect tests with both types.

File: ifrs17/Test/ImportStorageTest.ipynb [hunk 5]
```diff
@@ -390,7 +390,46 @@
...
+                "activity.Status.Should().Be(ActivityLogStatus.Succeeded);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "//There is a previous year, and for the current year Actuals are imported for a given Scenario",
```
Comment (dcolleoni) [inline ifrs17/Test/ImportStorageTest.ipynb:412]: Looking at this file, it is beneficial to separate each test case in a subSection and extract the little comment we have at the beginning as a little description. Thanks

File: ifrs17/Test/ImportStorageTest.ipynb [hunk 6]
```diff
@@ -390,7 +390,46 @@
...
+                "\n  };",
+                "\n",
+                "\nvar ivsBenchmark = new IfrsVariable[]{",
+                "\n    basicIfrsVariable with {Partition = partition.Id, AocType = AocTypes.BOP, EstimateType = EstimateTypes.DA, Value = 1000.0},",
+                "\n    basicIfrsVariable with {Partition = partition.Id, AocType = AocTypes.BOP, Value = 100.0},",
+                "\n    basicIfrsVariable with {Partition = partitionScenarioMTUP.Id, AocType = AocTypes.CL, Value = -15.0},",
+                "\n    basicIfrsVariable with {Partition = partitionScenarioMTUP.Id, AmountType = AmountTypes.CL, AocType = AocTypes.CL, Value = -99.0},",
+                "\n  };",
+                "\n",
+                "\nvar activity = await CheckIfrsVariablesFromImportStorageAsync(ImportFormats.Actual, inputDataSetForWorkspace, inputDataSetForDataSource, ivsBenchmark, argsScenarioMTUP);",
```
Comment (dcolleoni) [inline ifrs17/Test/ImportStorageTest.ipynb:432]: Interesting test cases are: 
Cashflow format for ScenarioX with primaryScope GricX with underlying GIC already imported BE (only and ScenarioX) too.
The Storage should contain IFRS of GIC (secondary scope) (of BE and ScenarioX, respectively).

File: ifrs17/Test/QueriesScenarioDataTest.ipynb [hunk 7]
```diff
@@ -0,0 +1,386 @@
...
+                "# Import"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Data"
```
Comment (dcolleoni) [inline ifrs17/Test/QueriesScenarioDataTest.ipynb:61]: These test could also contain interaction between primary and secondary import scope. 
At the moment I don't think QueryPartitionedDataAsync has any notion of the scope to query (it has only the partition). Correct?

Anyway, we plan to add a notion of import scope in #133. Can you add there a comment to add the above mention test case? thanks

File: ifrs17/Test/QueriesScenarioDataTest.ipynb [hunk 8]
```diff
@@ -0,0 +1,386 @@
...
+                "\nws.Initialize(x => x.FromSource(DataSource).DisableInitialization<RawVariable>().DisableInitialization<IfrsVariable>());"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var queriedRawVars = await ws.QueryPartitionedDataAsync<RawVariable,PartitionByReportingNodeAndPeriod>(DataSource, partitionScenarioMTUP.Id, partition.Id, ImportFormats.Cashflow);"
```
Comment (dcolleoni) [inline ifrs17/Test/QueriesScenarioDataTest.ipynb:204]: what if Scenario is YCU+10?
I expect, 
RawVariable to be BE from DataSource,
IfrsVariable (not BOP) to be BE from DataSource.
IfrsVariable (BOP) to be BE from DataSource period -1 even if there is a period -1 Scenario YCU+10 in the DataSource.

Comment (amuolo) [reply ifrs17/Test/QueriesScenarioDataTest.ipynb:204]: The case of scenario parameters (such as YCU+10) where BE Vars are taken from the DataSource is covered in the test 2.5 of this notebook. 

What do you mean with the last two lines about IfrsVar (not) BOP?

File: ifrs17/Test/QueriesScenarioDataTest.ipynb (hunk 7 above)
Comment (amuolo) [reply ifrs17/Test/QueriesScenarioDataTest.ipynb:61]: Ok, comment added.

File: ifrs17/Test/QueriesScenarioDataTest.ipynb (hunk 4 above)
Comment (amuolo) [reply ifrs17/Test/QueriesScenarioDataTest.ipynb:243]: done

File: ifrs17/Test/ImportStorageTest.ipynb (hunk 5 above)
Comment (amuolo) [reply ifrs17/Test/ImportStorageTest.ipynb:411]: done

**Review by dcolleoni — APPROVED** [review]

**Pull request merged**